src/server/api/routes/nft/mint-subscription.js

```diff
@@ -0,0 +1,145 @@
+const { Router } = require('express');
+const { firestore } = require('firebase-admin');
+const { v4: uuidv4 } = require('uuid');
+
+const { handleRestfulError } = require('../../middleware/error');
+
+const {
+  db,
+  nftMintSubscriptionCollection,
+} = require('../../../modules/firebase');
+
+const router = Router();
+
+router.post('/nft/mint-subscription', async (req, res, next) => {
+  const { wallet: subscribedWallet, email: subscriberEmail } = req.query;
+  if (!subscribedWallet) {
+    res.status(400).send('MISSING_SUBSCRIPTION_WALLET');
+    return;
+  }
+
+  if (!subscriberEmail) {
+    res.status(400).send('MISSING_SUBSCRIBER_EMAIL');
+    return;
+  }
+
+  const queryRef = nftMintSubscriptionCollection
+    .where('subscriberEmail', '==', subscriberEmail)
+    .where('subscribedWallet', '==', subscribedWallet)
+    .limit(1);
+  try {
+    await db.runTransaction(async t => {
+      const querySnapshot = await t.get(queryRef);
+      if (!querySnapshot.empty) {
+        res.status(409).send('ALREADY_SUBSCRIBED');
+        return;
+      }
+
+      const subscriptionId = uuidv4();
+      const docRef = nftMintSubscriptionCollection.doc(subscriptionId);
+      await t.set(docRef, {
+        subscriberEmail,
+        subscribedWallet,
+        ts: firestore.FieldValue.serverTimestamp(),
+      });
+      res.json({ subscriptionId });
+    });
+  } catch (err) {
+    handleRestfulError(req, res, next, err);
+  }
+});
+
+router.get('/nft/mint-subscription', async (req, res, next) => {
+  const { wallet: subscribedWallet, email: subscriberEmail } = req.query;
```

Comment:
We probably want to remove this endpoint after we have login sessions and user email in place. This is still kind of leak of information if I can just query status of an email + wallet pair

Reply:
Where is this api used actually?

Reply:
Removed